lib/LockService.js

```diff
@@ -0,0 +1,70 @@
+class LockoutTimeoutError extends Error {
+}
+
+class LockService {
+    constructor() {
+        this.queue = [];
+        this.locked = false;
+    }
+
+    async lock(callback, timeLimitMS = 500) {
```

Comment:
Some comments on what this service does would be helpful.
Locking a  callback is nothing intuitive to me. Shouldn't we be locking on some kind of object/value? Or is this a global lock?

Reply:
it's some kind of mutex for poor js devs.
There are some tests, on how to use it, but i can explain it here, and maybe we get a better name for it (it evolved over time)

- The LockService can be locked (not global, but per instance)
- Maybe we could rename it to Mutex/Sema, but lock is better known
-  it is a first in first out queue, where you only can start one "callback" at a time

example:
```
const lockService = new LockService()
const one = lockService.lock(() => { console.log('first'); await axios.get('someUrl); }) # directly executed 
const two = lockService.lock(() => { console.log('second') }) #  waiting for "first" to finish
```

node is single threaded, but can execute other code if the main thread is waiting (starting a promise, or calling a async function)

if there was no lock, the callback "second" would be executed after the console.log, since the code is waiting for axios.get() and the main thread would be execute other code.

Reply:
@Brutus5000 wanna rename it to mutex and aquire(callback, timeout)?

Reply:
Just to be clear is the callback the function that is run when the lock is acquired?

I think that is just where confusion might be coming from.

Reply:
yeah.
renamed all this to mutex->acquire